### PR TITLE
docs: add features of commands

### PR DESCRIPTION
**Pull request by ankithans**
Features we can focus on first https://github.com/aerogear/charmil/blob/suitable-cmds/docs/src/suitable_commands.md

#15

### CODE REVIEW

File: docs/src/suitable_commands.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Suitable commands
+
+### help
+
+### index
+
+- [x] Manage(list, add, delete) custom plugin indexes
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:7]: My take is that cli creators will hardcode their indexes. We can add that feature later.

We have two personas:

CLI creator (like ODO - who can use our SDK to bring some extensions they want to use)
CLI user (Person who uses ODO CLI)

My take is that indexes would be controlled by CLI creator and included at build time.

Comment (ankithans) [reply docs/src/suitable_commands.md:7]: You seem right, we can do this later. As other tasks look more important. As of now, we can have only one index and CLI developers can submit their plugins to that index.
later we can support their own hosted indexes

Comment (wtrocki) [reply docs/src/suitable_commands.md:7]: Not later. This practically doesn't make any sense to do. We are framework. Developers will add their indexes to code. What is the point of adding index command to their command line?

Comment (wtrocki) [reply docs/src/suitable_commands.md:7]: I knew this will sping some good conversation.

File: docs/src/suitable_commands.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+### info
+
+- [x] Show information about a plugin via NAME & INDEX/NAME
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:11]: Indexes would not be user facing - they will be part of the SDK IMHO
WDYT?>

File: docs/src/suitable_commands.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+### install - Downloads, Extract & Install a plugin
+
+- [x] Install one NAME or multiple plugins [Name, Name] from default index
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:15]: there will be no default index

File: docs/src/suitable_commands.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+- [ ] Install plugins from a newline-delimited txt file
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:16]: This is something needed. I think we need concept of profiles (they could have been hosted as kubernetes resources etc. )

Example: https://docs.openshift.com/container-platform/4.7/web_console/customizing-the-web-console.html#creating-custom-CLI-downloads_customizing-web-console

File: docs/src/suitable_commands.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- [x] Install one or mutiple plugins from custom index
+- [ ] Install plugin from a local/URL custom manifest file
+- [x] If a plugin is already installed, it will be skipped
+- [x] Failiure to install a plugin will not stop the installation of other plugins
+- [ ] Cleanup Stale Installations removes the versions that aren't the current version
+
+### list
+
+- [x] show list of all charmil plugins
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:25]: There is no concept of charmil plugins - Charmil is framework that let others to build their plugin indexes and embed them into their CLI. We need to be clear about this difference - I think I prematurely closed #7

File: docs/src/suitable_commands.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+### uninstall
+
+- [x] uninstall a plugin or list of plugins
+- [x] Failure to uninstall a plugin will result in an error and exit immediately
+- uninstall command does not support INDEX/PLUGIN syntax just specify PLUGIN
+
+### update
+
+- [x] update local copy of plugin indexes
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:40]: Not needed IMHO. That would be programmatic API instead.

File: docs/src/suitable_commands.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+### upgrade
+
+- [x] Install a plugin or all plugins to the newer version
+
+### version
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:46]: IMHO this is like status command. Could be useful and optional to include for SDK users (cli developers)

File: docs/src/suitable_commands.md (hunk 5 above)
Comment (ankithans) [reply docs/src/suitable_commands.md:25]: now it's all clear.

File: docs/src/suitable_commands.md (hunk 1 above)
Comment (wtrocki) [reply docs/src/suitable_commands.md:7]: > You seem right

btw.. I'm always wrong :) question everything I say.

Comment (ankithans) [reply docs/src/suitable_commands.md:7]: yup users of CLI don't need to know indexes and don't want to add/remove indexes. Its developer's headache.

Comment (ankithans) [reply docs/src/suitable_commands.md:7]: we are not having any single index, we provide freedom to the developer to host their indexes and include them. And for this purpose hardcoding suits.

File: docs/src/suitable_commands.md (hunk 2 above)
Comment (ankithans) [reply docs/src/suitable_commands.md:11]: users of CLI don't need to know indexes.

File: docs/src/suitable_commands.md (hunk 1 above)
Comment (wtrocki) [reply docs/src/suitable_commands.md:7]: Yes. to make it clear - multiple indexes as feature is good - but in SDK. This PR is all about end user Commands. 
There is separate issue to cover SDK #18

File: docs/src/suitable_commands.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# Suitable commands
```
Comment (wtrocki) [inline docs/src/suitable_commands.md:1]: To merge it we can:

- Move this document to `working-group` folder 
- Adjust it to reflect all commands we want to support and flag ones that will be done in the first place.

File: docs/src/suitable_commands.md (hunk 7 above)
Comment (ankithans) [reply docs/src/suitable_commands.md:46]: then not including this here, #18

File: docs/src/suitable_commands.md (hunk 8 above)
Comment (ankithans) [reply docs/src/suitable_commands.md]: `working-group`
is it a naming convention for what we are doing or in progress?

Comment (wtrocki) [reply docs/src/suitable_commands.md]: It is not docs - just research doc so use whatever name to reflect it

File: docs/src/suitable_commands.md (hunk 4 above)
Comment (ankithans) [reply docs/src/suitable_commands.md]: couldn't get this properly
Like in [OperatorHub](https://operatorhub.io/) Provider is the profile?
image

Comment (wtrocki) [reply docs/src/suitable_commands.md]: Yep :D

**Comment by ankithans** [comment]
now have a look is it fine?
https://github.com/aerogear/charmil/blob/suitable-cmds/working-group/suitable_commands.md

**Pull request merged**